### PR TITLE
remote config overhaul

### PR DESCRIPTION
**Pull request by benisgold**
Fixes APP-1040, APP-1041

## What changed (plus any additional context for devs)
The general problem: 
Remote config values are accessed from the `config` object in the app. The problem with this is that if the `config` object updates, it will not trigger rerenders in the components that statically access its attributes. This can cause race condition issues, since we do not await fetching remote config on app startup. 

How the points feature flag problem occurred:
1. swipe navigator that includes tab bar is rendered
2. remote config is fetched
3. since config is statically accessed, swipe navigator never rerenders with updated remote config value

Solution:
This PR migrates the remote config implementation over to a react query hook. This allows us to provide remote config with a hook, which causes dependent components to immediately rerender if remote config is updated. Now what happens:
1. swipe navigator that includes tab bar is rendered
2. remote config is fetched
3. since swipe navigator uses remote config query, it is rerendered using the appropriate remote config values

FOLLOW-UP:
* look into moving away from react query and using built-in firebase functionality

## Screen recordings / screenshots
https://www.loom.com/share/734dcab645144757814dc8b2e2334739

## What to test
TF 1.9.12 (28)
points feature should be hidden. try restarting the app many times to refresh remote config

### CODE REVIEW

File: src/navigation/SwipeNavigator.js [hunk 1]
```diff
@@ -80,6 +81,8 @@ const TabBar = ({
   const { width: deviceWidth } = useDimensions();
   const { colors, isDarkMode } = useTheme();
 
+  const data = useRemoteConfig();
+  console.log(data.test_do_not_use);
```
Comment (walmat) [inline src/navigation/SwipeNavigator.js:85]:
```suggestion
```
rm log

**Comment by walmat** [comment]
Should we move all consumption of the remote config to use the hook?

File: src/navigation/SwipeNavigator.js [hunk 2]
```diff
@@ -343,7 +353,9 @@ export function SwipeNavigator() {
 
   const [lastPress, setLastPress] = useState();
 
-  const showPointsTab = IS_DEV || IS_TEST || config.points_enabled;
+  const { points_enabled } = useRemoteConfig();
+
+  const showPointsTab = useExperimentalFlag(POINTS) || points_enabled;
```
Comment (walmat) [inline src/navigation/SwipeNavigator.js:358]: i think we had `IS_TEST` here before, right?

File: src/navigation/SwipeNavigator.js [hunk 3]
```diff
@@ -80,9 +81,11 @@ const TabBar = ({
   const { width: deviceWidth } = useDimensions();
   const { colors, isDarkMode } = useTheme();
 
-  const showPointsTab = IS_DEV || IS_TEST || config.points_enabled;
-  const NUMBER_OF_TABS = showPointsTab ? 4 : 3;
+  const { points_enabled } = useRemoteConfig();
+
+  const showPointsTab = useExperimentalFlag(POINTS) || points_enabled;
```
Comment (walmat) [inline src/navigation/SwipeNavigator.js:86]: > i think we had `IS_TEST` here before, right?

Also here

**Review by walmat — COMMENTED** [review]
testing the changes now

File: index.js [hunk 4]
```diff
@@ -25,6 +25,4 @@ to start before all of the imports.
  */
 require('react-native-gesture-handler');
 require('./shim');
-require('./src/model/config');
```
Comment (benisgold) [inline index.js:28]: is this ok @walmat

**Review by ibrahimtaveras00 — APPROVED** [review]
Tested turning remote config on and off for all respective parameters and works as expected; spot checked Android as well to make sure everything still shows!

QA Passed 👍🏽

**Pull request merged**